File: framework/java/src/main/java/soda/unittest/job/codec/IntArrayCodec.java
package soda.unittest.job.codec;

import java.util.List;

import soda.unittest.DataUtils;

public class IntArrayCodec implements ICodec<List<Integer>, int[]> {

	@Override
	public List<Integer> encode(int[] object) {
		return DataUtils.toList(object);
	}
	
	@Override
	public int[] decode(List<Integer> serial) {
		return DataUtils.toIntArray(serial);
	}
	
}

File: framework/scala/src/main/scala/soda/scala/unittest/codec/IntArrayCodec.scala
package soda.scala.unittest.codec

import soda.unittest.DataUtils

class IntArrayCodec extends ICodec[java.util.List[Integer], Array[Int]] {

    override def encode(obj: Array[Int]): java.util.List[Integer] = {
        DataUtils.toList(obj)
    }

    override def decode(ser: java.util.List[Integer]): Array[Int] = {
        DataUtils.toIntArray(ser)
    }

}

